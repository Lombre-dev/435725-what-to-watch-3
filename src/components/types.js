import PropTypes from 'prop-types';

export const Review = PropTypes.shape({
  id: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
});

export const Movie = PropTypes.shape({
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  year: PropTypes.number.isRequired,
  poster: PropTypes.string,
  frames: PropTypes.arrayOf(PropTypes.string).isRequired,
  preview: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  ratingReviewsCount: PropTypes.number.isRequired,
  description: PropTypes.string,
  story: PropTypes.string,
  duration: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string).isRequired,
  backgroundImage: PropTypes.string,
  backgroundColor: PropTypes.string,
  isFavorite: PropTypes.bool,
});
