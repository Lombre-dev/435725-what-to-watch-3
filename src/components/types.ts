import {AuthorizationStatus} from '../consts';

export type TReview = {
  id: number;
  author: string;
  score: number;
  text: string;
  date: number;
}

export type TReviewData = {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
};

export type TMovie = {
  id: number;
  title: string;
  genres: string[];
  year: number;
  poster?: string;
  frames: string[];
  preview: string;
  src: string;
  ratingScore: number;
  ratingReviewsCount: number;
  description?: string;
  story?: string;
  duration: number;
  director: string;
  actors: string[];
  backgroundImage?: string;
  backgroundColor?: string;
  isFavorite?: boolean;
}

export type TMovieData = {
  id: number;
  name: string;
  poster_image: string;
  preview_image: string;
  background_image: string;
  background_color: string;
  video_link: string;
  preview_video_link: string;
  description: string;
  rating: number
  scores_count: number;
  director: string;
  starring: string[];
  run_time: number;
  genre: string;
  released: number;
  is_favorite: boolean;
};

export type TUser = {
  id?: number;
  name?: string;
  email?: string;
  avatar?: string;
  authRequired?: boolean;
  authError?: number;
  authStatus?: AuthorizationStatus;
  favoriteMovies?: [];
}

export type TUserData = {
  id: number,
  email: string;
  name: string;
  avatar_url: string;
}

export type TMatchParamsWithId = {
  params: {
    id: string;
  }
}
