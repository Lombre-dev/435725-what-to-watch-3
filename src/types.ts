export type TReview = {
  id: number;
  author: string;
  score: number;
  text: string;
  date: number;
}

export type TReviewData = {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
};

export type TMovie = {
  id: number;
  title: string;
  genres: string[];
  year: number;
  poster?: string;
  frames: string[];
  preview: string;
  src: string;
  ratingScore: number;
  ratingReviewsCount: number;
  description?: string;
  story?: string;
  duration: number;
  director: string;
  actors: string[];
  backgroundImage?: string;
  backgroundColor?: string;
  isFavorite?: boolean;
}

export type TMovieData = {
  id: number;
  name: string;
  poster_image: string;
  preview_image: string;
  background_image: string;
  background_color: string;
  video_link: string;
  preview_video_link: string;
  description: string;
  rating: number;
  scores_count: number;
  director: string;
  starring: string[];
  run_time: number;
  genre: string;
  released: number;
  is_favorite: boolean;
};

export type TUser = {
  id?: number;
  name?: string;
  email?: string;
  avatar?: string;
  authRequired?: boolean;
  authError?: number;
  authStatus?: AuthorizationStatus;
  favoriteMovies?: [];
}

export type TUserData = {
  id: number;
  email: string;
  name: string;
  avatar_url: string;
}

export type TMatchParamsWithId = {
  params: {
    id: string;
  };
}

export enum AppPages {
  MAIN = `/`,
  LOGIN = `/login`,
  MY_LIST = `/mylist`,
  MOVIES = `/films`,
  PLAYER = `/player`,
}

export enum PlayerState {
  INITED = `inited`,
  LOADING = `loading`,
  PLAYING = `playing`,
  PAUSED = `paused`,
  ENDED = `ended`,
}

export enum ServerErrors {
  NOT_FOUND = 404,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
}

export enum AuthorizationErrorCode {
  INCORRECT_LOGIN = 1,
  AUTHORIZATION_FAIL = 2,
}

export enum AuthorizationStatus {
  NO_AUTH = `unauthorized`,
  AUTH = `authorized`,
}

export enum LoadingDataStatus {
  LOADING = `loading`,
  READY = `ready`,
  ERROR = `error`,
}
