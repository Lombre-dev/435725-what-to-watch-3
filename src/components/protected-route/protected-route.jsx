import PropTypes from 'prop-types';
import React from "react";
import {connect} from 'react-redux';
import {Redirect, Route} from 'react-router-dom';
import {AppPages} from '../../consts';
import {getUserAuthRequired} from '../../redux/user/selectors';

function ProtectedRoute({exact, path, render, authRequired}) {
  return (
    <Route
      exact={exact}
      path={path}
      render={() => {
        return (
          authRequired
            ? <Redirect to={AppPages.LOGIN} />
            : render()
        );
      }}
    />
  );
}

ProtectedRoute.propTypes = {
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  authRequired: PropTypes.bool,
};

function mapStateToProps(state) {
  return {
    authRequired: getUserAuthRequired(state),
  };
}

export {ProtectedRoute};
export default connect(mapStateToProps)(ProtectedRoute);
