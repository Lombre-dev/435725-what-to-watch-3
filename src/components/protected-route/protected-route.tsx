import * as React from 'react';
import {connect} from 'react-redux';
import {Redirect, Route} from 'react-router-dom';
import {getUserAuthRequired} from '../../redux/user/selectors';
import {AppPages} from '../../types';

type TProps = {
  exact: boolean;
  path: string;
  authRequired?: boolean;
  render: () => React.ReactNode;
};

function ProtectedRoute(props: TProps) {
  return (
    <Route
      exact={props.exact}
      path={props.path}
      render={() => {
        return (
          props.authRequired
            ? <Redirect to={AppPages.LOGIN} />
            : props.render()
        );
      }}
    />
  );
}

function mapStateToProps(state) {
  return {
    authRequired: getUserAuthRequired(state),
  };
}

export {ProtectedRoute};
export default connect(mapStateToProps)(ProtectedRoute);
