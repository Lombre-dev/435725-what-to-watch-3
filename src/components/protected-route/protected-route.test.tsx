import * as React from 'react';
import {BrowserRouter} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import {ProtectedRoute} from './protected-route';
import {AppPages} from '../../types';

const EXACT = true;
const PATH = AppPages.LOGIN;
const AUTH_REQUIRED = false;
const RENDER: () => React.ReactNode = () => {
  return <p>Test</p>;
};

describe(`<ProtectedRoute />`, () => {

  it(`render should be match markup`, () => {

    const result = renderer
      .create(<BrowserRouter>
        <ProtectedRoute
          exact={EXACT}
          path={PATH}
          authRequired={AUTH_REQUIRED}
          render={RENDER}
        />
      </BrowserRouter>, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
