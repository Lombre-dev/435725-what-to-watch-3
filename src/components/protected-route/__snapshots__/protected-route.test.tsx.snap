// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRoute /> render should be match markup 1`] = `null`;
